lib/nice_maps.ex

```diff
@@ -224,4 +224,76 @@ defmodule NiceMaps do
     key_type = Keyword.get(opts, :key_type)
     parse_key_type(key, key_type)
   end
+
+  @doc """
+  Merges the values of two given maps.
+
+  ## Options
+
+    - `:keys` (optional) only merge the given keys
+    - `:fun`  (optional) merge handler annonymous function that accepts two arguments
+
+  ## Examples
+
+      iex> acc_requests = %{success: ["200", "200", "201"], failed: []}
+      iex> new_requests = %{success: ["200", "200"], failed: ["404"]}
+      iex> NiceMaps.merge_values(acc_requests, new_requests)
+      %{success: ["200", "200", "201", "200", "200"], failed: ["404"]}
+
+      iex> acc_requests = %{success: ["200", "200", "201"], failed: []}
+      iex> new_requests = %{success: ["200", "200"], failed: ["404"]}
+      iex> NiceMaps.merge_values(acc_requests, new_requests, keys: [:success])
+      %{success: ["200", "200", "201", "200", "200"]}
+
+  """
+  @spec merge_values(map(), map(), keyword()) :: map()
+  def merge_values(%{} = map1, %{} = map2, opts \\ []) do
+    keys = Keyword.get(opts, :keys, Map.keys(map1))
+
+    Enum.reduce(keys, %{}, fn key, acc ->
+      Map.put_new(acc, key, apply_merge_fun(map1[key], map2[key], opts))
```

Comment:
We might need to expand on the merge function a little, to make it easier to handle various cases.

Maybe make it so that we can pass in merge handlers on a per-key basis?

So for example
```elixir
m1 = %{success: ["200", "200"], error: "NONE", number: 1}
m2 = %{success: ["201", "404", "505"], error: "Internal Server Error", number: 1}

error_merger = fn err1, err2 -> Enum.join([err1, err2], "\n") end
number_merger = fn n1, n2 -> n1 + n2 end

NiceMaps.merge_values(m1, m2, keys: [:success, error: error_merger, number: number_merger])
# => %{error: "NONE\nInternal Server Error", number: 3, success: ["200", "200", "201", "404", "505"]}
```